src/graphql/search/map/normalize.ts

```diff
@@ -0,0 +1,110 @@
+import queryString, { ParsedUrlQueryInput } from 'querystring'
+import { MAX_ZOOM, MIN_ZOOM, MapLayerType } from './config'
+import { MapCollection, MapCollectionLayer, LegendItemMapLayer } from '../../../generated/graphql'
+
+type InitialLegendItem = {
+  params: ParsedUrlQueryInput
+} & LegendItemMapLayer
+
+type InitialMapLayer = {
+  minZoom: number
+  maxZoom: number
+  params: ParsedUrlQueryInput
+  legendItems: Array<InitialLegendItem>
+} & MapCollectionLayer
+
+type InitialMapCollection = {
+  mapLayers: Array<InitialMapLayer>
+} & MapCollection
+
+export function composeId(collectionId: string, id: string) {
+  return `${collectionId}-${id}`
+}
+
+export function findMapLayer(mapLayers: Array<InitialMapLayer>, id: string) {
+  return mapLayers.find(mapLayer => mapLayer.id === id) || null
+}
+
+export function normalizeLegendItems(
+  collectionId: string,
+  legendItems: Array<InitialLegendItem>,
+  mapLayers: Array<InitialMapLayer>,
+) {
```

Comment:
This method is now missing the return type that was here before. I was expecting something like:

```ts
export function normalizeLegendItems(
  collectionId: string,
  legendItems: Array<InitialLegendItem>,
  mapLayers: Array<InitialMapLayer>,
): Array<LegendItemMapLayer> {  }
```